File: free-platform-book/free-platform-book-api/src/main/java/top/yangwulang/platform/entity/book/BookChapter.java
package top.yangwulang.platform.entity.book;

import io.swagger.v3.oas.annotations.media.Schema;
import org.babyfish.jimmer.Immutable;
import org.babyfish.jimmer.sql.*;

import javax.validation.constraints.NotNull;
import javax.validation.constraints.Null;

/**
 * 书籍章节
 */
@Entity
@Table(name = "book_chapter")
@Schema(description = "书籍章节")
@Immutable(value = Immutable.Nullity.NULLABLE)
public interface BookChapter {
    /**
     * 章节id
     */
    @Id
    @NotNull
    @Schema(description = "章节id")
    String id();

    /**
     * 章节标题
     */
    @Schema(description = "章节标题")
    String chapterTitle();

    /**
     * 书籍id
     */
    @ManyToOne
    @Key
    @OnDissociate(DissociateAction.DELETE)
    @Schema(description = "书籍信息", requiredMode = Schema.RequiredMode.REQUIRED)
    BookInfo book();

    /**
     * 来源
     */
    @Key
    @Schema(description = "书籍来源")
    String fromPath();
}

File: platform/src/main/kotlin/top/yangwulang/platform/entity/book/BookChapter.kt
package top.yangwulang.platform.entity.book

import org.hibernate.annotations.GenericGenerator
import java.io.Serializable
import javax.persistence.*

@Entity
@Table(name = "book_chapter")
open class BookChapter : Serializable {
    @Id
    @Column(name = "id", nullable = false, length = 64)
    @GeneratedValue(generator = "snowflake")
    @GenericGenerator(name = "snowflake", strategy = "top.yangwulang.platform.utils.SnowflakeGenerator")
    open var id: String? = null

    @Column(name = "chapter_title", length = 2000)
    open var chapterTitle: String? = null

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "book_id")
    open var book: top.yangwulang.platform.entity.book.BookInfo? = null

    @Column(name = "from_path", length = 2000)
    open var fromPath: String? = null
}